pause/pause.go

```diff
@@ -14,22 +13,37 @@
 // limitations under the License.
 //
 
-#include <signal.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <unistd.h>
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+func sigHandler(sig os.Signal) {
+	var ret int
 
-static void sigdown(int signo) {
-	psignal(signo, "shutting down, got signal");
-	exit(0);
+	switch sig {
+	case os.Interrupt:
+		fallthrough
+	case syscall.SIGTERM:
+		ret = 0
+
+	default:
+		ret = 42
+	}
+
+	fmt.Fprintf(os.Stderr, "shutting down, got signal %v\n", sig)
+	os.Exit(ret)
 }
 
-int main() {
-	if (signal(SIGINT, sigdown) == SIG_ERR)
-		return 1;
-	if (signal(SIGTERM, sigdown) == SIG_ERR)
-		return 2;
-	for (;;) pause();
-	fprintf(stderr, "error: infinite loop terminated\n");
-	return 42;
+func main() {
+	ch := make(chan os.Signal, 1)
+	signal.Notify(ch)
+
+	for {
```

Comment:
I think you could avoid to buffer the channel `ch` and that way, you don't need the loop `for`. The call to switch in the function signal handler should block until it gets a signal.